sentry-core/src/test/java/io/sentry/core/EnvelopeSenderTest.kt

```diff
@@ -0,0 +1,124 @@
+package io.sentry.core
+
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.argWhere
+import com.nhaarman.mockitokotlin2.eq
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.never
+import com.nhaarman.mockitokotlin2.times
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.whenever
+import java.io.File
+import java.io.FileNotFoundException
+import java.io.Reader
+import java.nio.file.Files
+import java.nio.file.Paths
+import kotlin.test.Test
+import kotlin.test.assertFailsWith
+import kotlin.test.assertFalse
+import kotlin.test.assertTrue
+
+class EnvelopeSenderTest {
+    private class Fixture {
+
+        var hub: IHub? = mock()
+        var envelopeReader: IEnvelopeReader? = mock()
+        var serializer: ISerializer? = mock()
+        var logger: ILogger? = mock()
+
+        init {
+            val options = SentryOptions()
+            options.isDebug = true
+            options.setLogger(logger)
+        }
+
+        fun getSut(): EnvelopeSender {
+            return EnvelopeSender(hub, envelopeReader, serializer, logger)
+        }
+    }
+
+    private val fixture = Fixture()
+
+    private fun getTempEnvelope(): String {
+        val testFile = this::class.java.classLoader.getResource("envelope-event-attachment.txt")
```

Comment:
I've created the class `FileFromResources` to make it easier, we can generalize it or create a new method in there to get the bytes

Reply:
Could you please clarify the value it adds? Here I'm just reading the resource in a single line. Not sure what `FileFromResources` is doing, it has a loop and adds line breaks? I wouldn't want the data to be modified.

Reply:
In this case, you just need the bytes to write on disk, so you are right and might not be suitable.
but the idea of the class would be a `Utils` for working with `files/json` content so we DRY, nothing more.
not sure why I've added `.append("\n")` either, maybe just for better visibility while debugging, can be safely removed as this would batter for envelopes.